### PR TITLE
[IN-825] session middleware - container for code review

### PR DESCRIPTION
**Pull request by ddombrow**
Note. There are a couple of skipped tests.

1. Cookie parser tests.  Basically these tests call cookieparser and then delete the cookie header out from under you so you have to rely on req.cookies or req.signedcookies. I don't think we need to support that...

2. The unsigned sid tests.  I can probably make these work and then we would get signature verification built in but I just wrote a test to make sure that we are signing properly and creating the extra cookie and left it at that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,10 +209,7 @@ parallel requests to your server and changes made to the session in one
 request may get overwritten when the other request ends, even if it made no
 changes (this behavior also depends on what store you're using).
 
-The default value is `true`, but using the default has been deprecated,
-as the default will change in the future. Please research into this setting
-and choose what is appropriate to your use-case. Typically, you'll want
-`false`.
+The default value is `false`.
```
Comment (danmactough) [inline README.md]: 😂

**Comment by ddombrow** [comment]
@danmactough for whatever reason the pillarjs/cookies module encodes the cookie string with all lowercase.  I assume in whatever RFC cookie spec it's not case sensitive so both are correct, but the tests checking for things in cookie strings all broke 😥

**Comment by danmactough** [comment]
Makes sense: https://tools.ietf.org/html/rfc2109

```
4.1  Syntax:  General

   The two state management headers, Set-Cookie and Cookie, have common
   syntactic properties involving attribute-value pairs.  The following
   grammar uses the notation, and tokens DIGIT (decimal digits) and
   token (informally, a sequence of non-special, non-white space
   characters) from the HTTP/1.1 specification [RFC 2068] to describe
   their syntax.

   av-pairs        =       av-pair *(";" av-pair)
   av-pair         =       attr ["=" value]        ; optional value
   attr            =       token
   value           =       word
   word            =       token | quoted-string

   Attributes (names) (attr) are case-insensitive.  White space is
   permitted between tokens.  Note that while the above syntax
   description shows value as optional, most attrs require them.

   NOTE: The syntax above allows whitespace between the attribute and
   the = sign.
```

**Comment by ddombrow** [comment]
so yeah, two valid implementations, it was only the tests that were case-sensitive

**Pull request merged**